### ISSUE TITLE
Allow NiftiMapsMasker.fit_transform to take a list of images as input

### ISSUE THREAD

**Issue by koudyk** [nilearn/nilearn#2247]
**What would you like changed/added and why?**
Allow NiftiMapsMasker.fit_transform to take a list of images as input, just like how ConnectivityMeasure.fit_transform can take a list of inputs.  

**What would be the benefit? Does the change make something easier to use?
Clarifies something?**
It would make the functions more consistent, and it might be easier for some users to input a list instead of building their own for loop.

**Comment by jeromedockes** [comment]
Thanks! could you provide a small example showing what is the problem? Passing a
list of images to `fit_transform` seems to work for me, as well as passing the
atlas to `__init__` as a list.

for example does this work on your machine?:

```python
from nilearn import datasets, input_data, image

atlas = datasets.fetch_atlas_msdl()
masker = input_data.NiftiMapsMasker(atlas["maps"])
mni = datasets.load_mni152_template()
masked = masker.fit_transform([mni, mni, mni])
print(masker.maps_img_.shape)
print(masked.shape)

atlas_maps = list(image.iter_img(atlas["maps"]))
print(atlas_maps[:3])
masker = input_data.NiftiMapsMasker(atlas_maps)
mni = datasets.load_mni152_template()
masked = masker.fit_transform([mni, mni, mni])
print(masker.maps_img_.shape)
print(masked.shape)
```

**Comment by koudyk** [comment]
Thanks for the quick reply! 
The code you gave does work on my machine, but my code still does not work. The difference that stands out to me is that I give a list of confounds as well as a list of images. Is it possible that the function can't take a list of confounds? I can't tell from the docs or the code (probably because of my lack of experience more than the lack of clarity). 

Here is my code:
```
from nilearn.input_data import NiftiLabelsMasker
from nilearn import datasets

development_dataset = datasets.fetch_development_fmri(n_subjects=90)
multiscale = datasets.fetch_atlas_basc_multiscale_2015(version='sym')
atlas_filename = multiscale.scale064
masker = NiftiLabelsMasker(labels_img=atlas_filename, standardize=True,
                           memory='nilearn_cache', verbose=0)
time_series_all_sub = masker.fit_transform(development_dataset.func, confounds=development_dataset.confounds)
```

**Comment by KamalakerDadi** [comment]
maps masking is done on 4D as input.

list of 3D is still a 4D. thats why Jerome code works. But if you provide list of 4D, it should not work. I think this is what kendra is pointing at. If I get this issue correctly.

for example, the existing should point to it #1240

**Comment by Hororohoruru** [comment]
@koudyk are you by chance working on this already? I was considering open a PR to address it implementing the changes suggested in #1240

**Comment by koudyk** [comment]
@Hororohoruru no, I'm not working on this already. Sorry for the delayed response!

**Comment by Remi-Gau** [comment]
Will close due to the lack of activity and lack of code to reproduce this issue.

Please reopen if you are facing this issue.
